guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,398 @@
+"""
+Title: CutMix, MixUp, and RandAugment image augmentation with KerasCV
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/04/08
+Last modified: 2022/04/08
+Description: Use KerasCV to augment images with CutMix, MixUp, RandAugment, and more!
+"""
+
+"""
+## Overview
+
+KerasCV makes it easy to assemble state of the art, industry grade data augmentation
+pipelines for image classification and object detection tasks.  KerasCV offers a wide
+suite of preprocessing layers implementing common data augmentation techniques.
+
+Perhaps three of the most useful layers are `CutMix`, `MixUp`, and `RandAugment`.  These
+layers are used in nearly all state of the art image classification pipelines.
+
+This guide will show you how to compose these layers into your own state of the art data
+augmentation pipeline for image classification tasks.  This guide will also walk you
+through the process of customizing a KerasCV data augmentation pipeline.
+"""
+
+"""
+## Imports and Setup
+
+This tutorial requires you to have KerasCV installed:
+
+```shell
+!pip install keras-cv
+```
+
+We will also begin by import all required packages:
+"""
+
+import keras_cv
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.keras import applications
+from tensorflow.keras import losses
+from tensorflow.keras import optimizers
+
+"""
+## Data Loading
+
+This guide uses the oxford_flowers102 dataset for demonstration purposes.
+
+To get started, we will first load the dataset:
+"""
+
+AUTOTUNE = tf.data.AUTOTUNE
+tfds.disable_progress_bar()
+data, dataset_info = tfds.load("oxford_flowers102", with_info=True, as_supervised=True)
+
+"""
+Next, we resize the images to a constant size, `(224, 224)`, and one hot encode the
+labels.  Please note that `keras_cv.layers.CutMix` and `keras_cv.layers.MixUp` expect
+`labels` to be one hot encoded.  This is because they modify the values held in `labels`
+in a way that is not possible with a sparse label representation.
+"""
+
+IMAGE_SIZE = (224, 224)
+BATCH_SIZE = 32
+num_classes = dataset_info.features["label"].num_classes
+
+
+def prepare(image, label):
+    image = tf.image.resize(image, IMAGE_SIZE)
+    image = tf.cast(image, tf.float32)
+    label = tf.one_hot(label, num_classes)
+    return {"images": image, "labels": label}
+
+
+def prepare_dataset(dataset, split):
+    if split == "train":
+        return (
+            dataset.map(lambda x, y: prepare(x, y), num_parallel_calls=AUTOTUNE)
```

Comment:
Why do we need to lambda it? The data is already generating the images and the labels I think. If that's the case, we should be able to:

`dataset.map(prepare, num_parallel_calls=AUTOTUNE)`

Reply:
Done